### PR TITLE
PAYEE-990 - Adding last signed in functionality

### PR DESCRIPTION
**Pull request by brendanstraughan**
- [ ] Unit tests passing
- [ ] Coverage reached
- [ ] Acceptance tests passing
- [ ] Reviewed

### CODE REVIEW

File: app/controllers/auth/AuthedUser.scala [hunk 1]
```diff
@@ -34,21 +36,24 @@ object AuthedUser {
     nino: Option[String],
     saUtr: Option[String],
     providerType: Option[String],
-    confidenceLevel: ConfidenceLevel): AuthedUser = {
+    confidenceLevel: ConfidenceLevel,
+    loginTimes: LoginTimes): AuthedUser = {
     val validNino = nino.getOrElse("")
-    AuthedUser(validNino, saUtr, providerType, confidenceLevel, None)
+    AuthedUser(validNino, saUtr, providerType, confidenceLevel, None, loginTimes)
```
Comment (sergi-hmrc) [inline app/controllers/auth/AuthedUser.scala:42]: valid nino is the absence of nino?!

File: app/uk/gov/hmrc/tai/util/DateHelper.scala [hunk 2]
```diff
@@ -43,4 +45,12 @@ object DateHelper {
     monthAndYearRegex.findFirstIn(date).getOrElse("")
   }
 
+  def dateTimeFormat(dt: DateTime): String = {
+    val month = dt.monthOfYear.getAsText
+    val day = dt.getDayOfMonth
+    val year = dt.getYear
+    val time = dt.toLocalTime.toString("HH:mm")
+    s"$day $month $year at $time"
```
Comment (sergi-hmrc) [inline app/uk/gov/hmrc/tai/util/DateHelper.scala:53]: would something like this ```DateTimeFormatter.ofPattern("dd MMM yyyy 'at' HH:mm")``` work?

Comment (sergi-hmrc) [reply app/uk/gov/hmrc/tai/util/DateHelper.scala:53]: i see it's joda time in that case: 
```def dateTimeFormat(dt: DateTime): String = dt.toLocalTime.toString(DateTimeFormat.forPattern("dd MMM yyyy 'at' HH:mm"))```
also add a test with some DateTime that checks it follows the string pattern

Comment (sergi-hmrc) [reply app/uk/gov/hmrc/tai/util/DateHelper.scala:53]: https://stackoverflow.com/questions/63737832/datetime-format-needed-chnages-as-expected

**Review by sergi-hmrc — APPROVED** [review]

**Review by masahmad — APPROVED** [review]

**Review by vivekhmrc — APPROVED** [review]

**Pull request closed without merging**